### ISSUE TITLE
<head /> tags from "next/head" not rendered in production

### ISSUE THREAD

**Issue by duncan-meech** [vercel/next.js#24794]
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

12.18.2

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Custom NextJS server

### Describe the Bug

At DoorDash we are starting to rollout pages to production using next-js and deployed into our own server infra running a custom nodejs/express/next-js app.

We are blocked by an issue with <head /> element created from "next/head".
Running the development build locally OR the optimized production build locally we see both the <Head /> inserted by _document.tsx AND elements ( e.g. title, description ) add to a <head /> element rendered within the particular page that is rendering.

However when deployed to production to <head /> tags introduced at the page level are absent.
Locally we see `<meta name="next-head-count" content="16" />`
In production we see `<meta name="next-head-count" content="16" />`

We are stumped by this non-deterministic behavior.
**NOTE**
After the page has loaded the HTML has been rendered client side we DO see the <head /> elements as expect. But they are not part of the initial HTML response sent from the server.

### Expected Behavior

<head /> elements are rendered identically for local dev, optimized production and deployed optimized production builds.

**Production build locally**
<img width="782" alt="Screen Shot 2021-05-04 at 2 03 13 PM" src="https://user-images.githubusercontent.com/66681819/117069570-7fff7800-ace1-11eb-8e9b-fe40d5372e2c.png">

**Production build deployed**
<img width="665" alt="Screen Shot 2021-05-04 at 2 04 17 PM" src="https://user-images.githubusercontent.com/66681819/117069689-a6bdae80-ace1-11eb-9ee6-c549b8693900.png">

### To Reproduce

**_document.tsx**
```
  render() {
    // the locale is provided by the next router which is pre-configured for certain locale prefixes
    const { locale } = this.props.__NEXT_DATA__
    // For now we only support en-US on SSR, but this will need to be dynamic
    return (
      <Html lang={locale}>
        <Head>
          <link key="favicon" rel="icon" href="/favicon.ico" />
          <meta key="ssr" name="ssr" content="true" />
          {this.renderSegmentSnippet()}
          <script key="google" src={googleMapsSnippet} />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
```
**example page tsx ( how we render the head per page )**
```
export function SSRHead({
  language,
  pageTitle,
  pageDescription,
  originalURL,
}: Props) {
  return (
    <Head>
      {/* used to initialize the language for SSR */}
      <meta name="dd_locale" content={language} />
      {/* page title and description */}
      <title>{pageTitle}</title>
      <meta name="description" content={pageDescription || pageTitle} />
      {/* x-language tags derived from current language URL */}
      {HrefLang(originalURL)}
    </Head>
  )
}
```

**Comment by timneutkens** [comment]
Please add a complete reproduction as asked for in the issue template, it's hard to help based on excerpts from 2 files.

**Comment by wholebuzz** [comment]
Exact same experience here.  Haven't been able to determine what the difference is.  Did you figure anything else out?

**Comment by wholebuzz** [comment]
The basic difference seems to be the operating system.  For an app built on macos and running on macos <Head> always works for me.  But built on linux or running on linux doesn't.  There's actually one page that <Head> works on for Linux but can't understand why.

**Comment by wholebuzz** [comment]
Ahh and underlying all of this is Error: only absolute urls are supported.  Causing initialProps to be different and <Head> not called.  So not even a problem with Head per se, but the likely place many different errors will manifest.

**Comment by timneutkens** [comment]
Closing as no reproduction was provided.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
